Implement fix for UI bug regarding addRole command

kerbysoh opened the pull request:
- Altered the listener on the UI component for the list of roles tagged to a company to refresh the list of roles whenever there is a change to the list.

kerbysoh committed:
Implement fix for UI bug regarding addRole command

- Altered the listener on the UI component for the list of roles tagged to a company to refresh the list of roles whenever there is a change to the list.

petermonky commented:
Just tried the fix but there seems to be an issue. The UI updates for when the company initially has a populated role list since the start of the application, but does not update otherwise. This issue can be recreated by calling `addRole` with a company that has an empty role list from the start.

kerbysoh commented:
Thanks for the find!

kerbysoh committed:
Implement additional fix for UI Bug regarding addRole command

- Abstracted the updating of the roleListPanelPlaceholder into a function

- Alter roleListPanel whenever roleList is altered and updates the roleListPanelPlaceholder with the new roleListPanel

petermonky commented:
LGTM 👍 nice use of SLAP too.
